### PR TITLE
Merge Speed up and Refactoring

### PR DESCRIPTION
**Pull request by kim-mskw**
This pull request introduces several updates and improvements across multiple data pipeline and training scripts, primarily to enhance configuration flexibility, streamline data handling, and improve model training workflows. The most significant changes include consistent use of start/end date configuration, new data preprocessing and postprocessing steps, and enhancements to the model training script to support resuming from checkpoints and improved logging.

**Configuration and Data Handling Updates:**

- Standardized the use of `DATA_START` and `DATA_END` variables (from `src.shared.config`) in place of the previous `START` and `END` throughout data acquisition and transformation scripts, ensuring a consistent and configurable approach to specifying data ranges. 

- Improved environment variable loading by using `find_dotenv()` and setting `override=True` to ensure correct environment configuration.

**Data Processing and Preprocessing Enhancements:**

- Added new postprocessing steps in `01b_data_acquisition.py`:
    - Rebuilt and merged ID-full data using `run_build_idfull_and_merge`.
    - Introduced a precompute step for VWAP matrices (required for downstream algorithms) with `precompute_range`, saving results to a configurable path. 

- Updated data transformation logic to use dynamically generated file names based on configuration dates and removed hardcoded date ranges. 

**Simulation and Model Training Improvements:**

- Modified simulation scripts to reference new or updated simulation functions, such as switching to `simulate_period` from `rolling_intrinsic_gurobi_qh` and updating the multi-market simulation to use `simulate_days_stacked_quarterhourly_products`. 

- Enhanced the training script for coordinated multi-market models:
    - Added support for resuming training from a specified checkpoint, including logic to handle versioned output directories.
    - Improved device selection and logging.
    - Updated model instantiation to include new parameters (e.g., `reward_log_path`) and adjusted hyperparameters.
    - Incorporated more robust handling of input data splits and preprocessing. 

**Other Notable Changes:**

- Cleaned up and clarified code, such as removing unused variables, commenting out legacy code, and improving comments for maintainability. 

These updates collectively improve the maintainability, configurability, and robustness of the data pipeline and model training workflow.

### CODE REVIEW

File: 01b_data_acquisition.py [hunk 1]
```diff
@@ -115,7 +115,7 @@
 csv_path = Path(output_path, f"data_{data_start}_{data_end}_hourly.csv")
 data_hourly.to_csv(csv_path)
 
-
+# TODO: Das brauchst du hier nciht korrekt?
```
Comment (TimFsl) [inline 01b_data_acquisition.py]: Ja das kann raus

File: 01b_data_acquisition.py [hunk 2]
```diff
@@ -125,6 +125,7 @@
 # You can adjust bucket size and min trades as needed
 # Matrices are required for Rolling Intrinsic Algortihm
 
+#TODO: Hier nicht ordentlich die config benutzt und das macht keinen Sinn das als Funktion zu definieren.
```
Comment (TimFsl) [inline 01b_data_acquisition.py]: ja das stimmt. Ich hab das angepasst in config und dem runner und unter main in precompute_vwaps die Importe aus config hinzugefügt falls man nur das script laufen lassen will

File: 04b_train_coordinated_multi_market.py [hunk 3]
```diff
@@ -147,6 +147,7 @@
             n_steps=512,
             clip_range=0.4,
             batch_size=128,
+            # TODO: wieso hast du das Gewicht von dem value loss kleiner gemacht?
```
Comment (TimFsl) [inline 04b_train_coordinated_multi_market.py]: Ich habe mit den ganzen Parametern rumgespielt und getestet, der Wert hat keine spezielle Bedeutung

File: 02b_single_market_rolling_intrinsic_qh.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 warnings.simplefilter(action="ignore", category=FutureWarning)
 
 if __name__ == "__main__":
-
+# TODO: Du hattest vorher die Bucket size in dem vWAp pre calculation neu gesetzt. man sollte hier keinen Möglichkeit haben die zu überschrieben sondern sich die aus dem precalculated file holen. Überlege mal ob du das alt meta da mit ablegen kannst. Ansonsten führt das nur zu komischen Fehlern.
```
Comment (TimFsl) [inline 02b_single_market_rolling_intrinsic_qh.py]: Ja ergibt Sinn. Ich habe jetzt bei allen drei RI Skripten eine neue mini Funktion eingebaut welche aus den VWAPs Matrizen über die Indizes die Größe ableitet. Die wird dann innerhalb der ersten while Schleife aufgerufen und so die bucket size bestimmt.

File: src/coordinated_multi_market/rolling_intrinsic/testing_rolling_intrinsic_qh_intelligent_stacking.py [hunk 5]
```diff
@@ -190,6 +190,7 @@ def build_battery_model(
     c_rate: float,
     roundtrip_eff: float,
 ) -> Tuple[gp.Model, Dict[str, Any], Dict[Any, gp.Constr], gp.Constr]:
+    # TODO: gleiches hier hast du verifiziert, dass das Model vorher und nachher gleich funktioniert hat?
```
Comment (TimFsl) [inline src/coordinated_multi_market/rolling_intrinsic/testing_rolling_intrinsic_qh_intelligent_stacking.py]: Ja, die Modelle liefern bis auf minimale numerische Abweichungen dieselben Ergebnisse.

File: src/coordinated_multi_market/basic_battery_dam_env.py [hunk 6]
```diff
@@ -182,6 +182,7 @@ def step(
             # check if we have capacity left in battery
             if self._current_soc > 0:
                 # penalty because missed profit
+                # TODO: Ahhhh, ja wie sieht deine penalty im Vergleich zum Reward denn jetzt aus? Macht ja Sinn das der sich anders verhält im Vergleich zu meinem Training. Deine peanlty ist jetzt ja viel kleiner.
```
Comment (TimFsl) [inline src/coordinated_multi_market/basic_battery_dam_env.py]: Ja hier gilt dasselbe wie bei den Parametern, ich habe mit dem Reward Sharing rumgespielt und Sachen ausprobiert. Besser noch nicht übernommen sondern einfach einmal final wenn das Modell sauber läuft

File: src/coordinated_multi_market/learning_utils.py [hunk 7]
```diff
@@ -13,6 +13,7 @@
 train_end = DATA_END.date().isoformat()
 path = f"df_spot_train_{train_start}_{train_end}_with_features_utc.csv"
 
+# TODO: Finde ich nicht gut. Würde dir empfehlen auf das alte Sampling zu ändern.
```
Comment (TimFsl) [inline src/coordinated_multi_market/learning_utils.py]: Lass uns dazu am Montag nochmal reden

File: src/data_acquisition/epex_sftp/build_idfull.py [hunk 8]
```diff
@@ -110,7 +110,7 @@ def merge_idfull_into_csv(csv_in: str, csv_out: str = None) -> pd.DataFrame:
 
     if "id_full_h" in df.columns:
         df.drop(columns=["id_full_h"], inplace=True)
-        
+    #TODO: Benutzen das später für den Spread. Evt kann man da noch andere Spreads nehmen. Zum Beispiel den ID3 für das Produkt, weil das wird häufig als proxy für den Kauf/Verkauf auf dem ID genommen.
```
Comment (TimFsl) [inline src/data_acquisition/epex_sftp/build_idfull.py]: Verstehe den Kommentar nicht ganz. Meinst du ich soll auch noch die anderen Indizes berechnen wie ID3 ?

File: src/coordinated_multi_market/custom_ppo.py [hunk 9]
```diff
@@ -177,6 +168,7 @@ def collect_rollouts(
             # --- DA-Teil: immer vorhanden ---
             da_rewards = rollout_buffer.rewards[row_start : row_start + num_rows].flatten()
 
+            # TODO: Wenn du es immer berechnest warum ist es dann optional?
```
Comment (TimFsl) [inline src/coordinated_multi_market/custom_ppo.py]: Ja der Kommentar kann weg

File: src/coordinated_multi_market/custom_ppo.py [hunk 10]
```diff
@@ -422,12 +411,10 @@ def _check_if_complete_cycle(period_volumes, capacity: float = 1.0,
                              min_cycle_fraction: float = 1.0):
         traded_volume = abs(period_volumes).sum()
         cycle_fraction = traded_volume / (2 * capacity)
+        
+        # TODO: Wieso hier größer gleich? IN welchem Case erlaubst du mehr als 1 Cycle?
```
Comment (TimFsl) [inline src/coordinated_multi_market/custom_ppo.py]: Das ganze ist nur relevant wenn man den Wert von min_cycle_fraction auf kleiner als 1 setzt. Daher das größer gleich Operator. Ich hatte die Funktion umgeändert damit es möglich ist dass der Agent eine offene Position eingehen kann und trotzdem RI-Reward bekommt. Wenn das auf 1 ist muss er ja einen vollständigen Cycle im DayAhead machen um überhaupt RI-Reward zu bekommen. Wenn man den min_cycle_fraction Wert auf etwas zwischen 0 bis 1 setzt hat er dann mehr Flexibilität

File: src/single_market/rolling_intrinsic_gurobi_qh.py [hunk 11]
```diff
@@ -562,6 +562,7 @@ def simulate_period(
         logger.info(f"Trading end:   {trading_end}")
 
         # Battery time index (delivery quarter-hours)
+        # TODO: warum so kompliziert?
```
Comment (TimFsl) [inline src/single_market/rolling_intrinsic_gurobi_qh.py]: habe ich angepasst.

File: src/data_acquisition/epex_sftp/precompute_vwaps.py [hunk 12]
```diff
@@ -129,17 +126,19 @@ def precompute_vwaps_for_day(
     )
 
     # All quarter-hourly products for this delivery day
+    # TODO: Warum ist das so umständlich definiert? start und end of day sind doch eh fixe Werte?
```
Comment (TimFsl) [inline src/data_acquisition/epex_sftp/precompute_vwaps.py]: Habe ich angepasst
